CodeGeneration/Sources/SyntaxSupport/KeywordSpec.swift

```diff
@@ -0,0 +1,291 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public struct ContextualKeywordSpec {
+  public var name: String
+  public var isLexerClassified: Bool
+  public var requiresLeadingSpace: Bool
+  public var requiresTrailingSpace: Bool
+
+  public var escapedName: String {
+    if isLexerClassified || name == "Type" || name == "Protocol" {
+      return "`\(name)`"
+    } else {
+      return name
+    }
+  }
+
+  init(_ name: String, isLexerClassified: Bool = false, requiresLeadingSpace: Bool = false, requiresTrailingSpace: Bool = false) {
```

Comment:
`isLexerClassified` could use some documentation/a different name. I haven't thought of something better yet though. Maybe something with "strong" in 🤔

Reply:
Good idea, I just copied the documentation from the `isLexerClassified` property on `Keyword`.

I’m not super proud of the name but at least `isLexerClassified` exactly describes what the property does whereas anything like “strong” is a lot more ambiguous, so I’d prefer to stick to it.